.github/workflows/ci.yml

```diff
@@ -464,6 +464,69 @@ jobs:
       # TODO: Do this check on Windows too.
       - run: mk/check-symbol-prefixes.sh --target=${{ matrix.target }}
 
+  # The wasm32-wasi targets have a different set of feature sets.
+  test-wasm32-wasi:
+    # Don't run duplicate `push` jobs for the repo owner's PRs.
+    if: github.event_name == 'push' || github.event.pull_request.head.repo.full_name != github.repository
+
+    runs-on: ${{ matrix.host_os }}
+
+    strategy:
+      matrix:
+        host_os:
+          - ubuntu-22.04
+        mode:
+          - # debug
+          - --release
+        rust_channel:
+          - stable
+          - beta
+          - nightly
+        target:
+          - wasm32-wasi
+
+    steps:
+      - if: ${{ contains(matrix.host_os, 'ubuntu') }}
+        run: sudo apt-get update -y
+
+      - uses: briansmith/actions-checkout@v2
+        with:
+          persist-credentials: false
+
+      - run: cargo generate-lockfile
+
+      - run: mk/install-build-tools.sh --target=${{ matrix.target }} ${{ matrix.features }}
+
+      - uses: briansmith/actions-rs-toolchain@v1
+        with:
+          override: true
+          target: ${{ matrix.target }}
+          toolchain: ${{ matrix.rust_channel }}
+
+      - name: Install wasi-sdk
+        run: |
+          export WASI_URL=$(curl -s https://api.github.com/repos/WebAssembly/wasi-sdk/releases/latest | jq -r '.assets[] | select(.name | contains("linux.tar.gz")) | .browser_download_url')
+          wget -O ./wasi-sdk-linux.tar.gz $WASI_URL
+          mkdir wasi-sdk
```

Comment:
Actually, why do we need the WASI SDK? If you look at the existing wasm32 targets, we manage to build them with standard clang, and I don't see anything WASI-specific here besides the rand.rs, which doesn't add any new C stuff. Maybe we can get rid of this SDK dependency by treating the `-wasi` target just like the `-unknown-unknown` target?